keps/104-StartupPolicy/README.md

```diff
@@ -0,0 +1,420 @@
+highlighting any additional information provided beyond the standard KEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API Proposal](#api-proposal)
+  - [Implementation](#implementation)
+  - [Validation](#validation)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+  - [Poor Person Workflow Engine](#poor-person-workflow-engine)
+  - [Increase number of patches to jobset](#increase-number-of-patches-to-jobset)
+- [Alternatives](#alternatives)
+  - [Operate on Conditions](#operate-on-conditions)
+  - [Allow for both ready and succeeded](#allow-for-both-ready-and-succeeded)
+<!-- /toc -->
+
+## Summary
+
+This KEP adds a StartupPolicy for Jobsets.  A StartupPolicy allows for some flexibility on when to start ReplicatedJobs in a Jobset.  
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+High Performance Computing / Machine Learning users usually want some control over how jobs are being created.  A Startup policy would allow a user to specify which status a replicated job should have before starting other replicated jobs in a jobset.
+
+### Goals
+
+- Add a startup policy for JobSet
+
+### Non-Goals
+
+- Complicated workflow specifications
+  - This means that we will support simple dependecies rather than processing a full dag.
```

Comment:
```suggestion
  - This means that we will support simple dependecies rather than processing a full DAG.
```
And maybe we could better clarify this - one level of dependencies? One level of flat start up order logic? I don't think we want to imply that we are even processing a DAG here to keep the idea of a workflow entirely separate. This is just a startup policy, no DAG or workflow relevancy. It's for a step in a workload. https://kubernetes.io/docs/concepts/workloads/ (I assume JobSet would eventually be added there).